cmd/kubeadm/app/util/endpoint.go

```diff
@@ -36,16 +37,25 @@ func GetMasterEndpoint(cfg *kubeadmapi.MasterConfiguration) (string, error) {
 	return fmt.Sprintf("https://%s", hostPort), nil
 }
 
-// GetMasterHostPort returns a properly formatted Master IP/port pair or error
-// if the IP address can not be parsed or port is outside the valid TCP range.
+// GetMasterHostPort returns a properly formatted Master hostname or IP and port pair, or error
+// if the hostname or IP address can not be parsed or port is outside the valid TCP range.
 func GetMasterHostPort(cfg *kubeadmapi.MasterConfiguration) (string, error) {
 	var masterIP string
+	var portStr string
 	if len(cfg.API.ControlPlaneEndpoint) > 0 {
-		errs := validation.IsDNS1123Subdomain(cfg.API.ControlPlaneEndpoint)
+		if strings.Contains(cfg.API.ControlPlaneEndpoint, ":") {
+			var err error
```

Comment:
This doesn't need to be explicit.

Reply:
You could use implicit declaration below.

Reply:
Then masterIP and portStr would be shadowed, that's why I explicitly declared the error.